src/Compilers/CSharp/Portable/Syntax/SyntaxNodeExtensions.cs

```diff
@@ -128,15 +133,25 @@ internal static bool IsVariableDeclarationInitialization(this SyntaxNode node)
                 return false;
             }
 
-            return variableDeclarator.Parent.IsKind(SyntaxKind.VariableDeclaration);
+            SyntaxNode variableDeclaration = variableDeclarator.Parent;
+            if (!variableDeclaration.IsKind(SyntaxKind.VariableDeclaration))
+            {
+                return false;
+            }
+
+            return
```

Comment:
>return [](start = 12, length = 6)

Could you explain the tightening here? Is it observable?

Reply:
It isn't observable in code that was previously valid. Now that we are making it valid, we need the tightening to narrow the places where we consider stackalloc to be a pointer to just those places where it was previously valid *and* a pointer.

---
In reply to: [207691737](https://github.com/dotnet/roslyn/pull/28969#discussion_r207691737) [](ancestors = 207691737)